sdk/@launchdarkly/observability-node/src/client/handlers.ts

```diff
@@ -0,0 +1,226 @@
+import type { Attributes } from '@opentelemetry/api'
+import * as http from 'http'
+import { NodeOptions } from '../api/Options.js'
+import {
+	ATTR_HTTP_REQUEST_METHOD,
+	ATTR_HTTP_ROUTE,
+	ATTR_HTTP_RESPONSE_STATUS_CODE,
+} from '@opentelemetry/semantic-conventions'
+import { _LDObserve } from '../sdk/LDObserve.js'
+
+/** JSDoc */
+interface MiddlewareError extends Error {
+	status?: number | string
+	statusCode?: number | string
+	status_code?: number | string
+	output?: {
+		statusCode?: number | string
+	}
+}
+
+function processErrorImpl(
+	options: NodeOptions,
+	req: { headers?: http.IncomingHttpHeaders },
+	error: Error,
+	metadata?: Attributes,
+): void {
+	const { secureSessionId, requestId } = _LDObserve.parseHeaders(
+		req.headers ?? {},
+	)
+	_LDObserve._debug('processError', 'extracted from headers', {
+		secureSessionId,
+		requestId,
+	})
+
+	_LDObserve.recordError(error, secureSessionId, requestId, metadata)
+	_LDObserve._debug('consumed error', error)
+}
+
+/**
+ * Express compatible middleware.
+ * Exposed as `Handlers.errorHandler`.
+ * `metadata` accepts structured tags that should be attached to every error.
+ */
+export function middleware(
+	metadata?: Attributes,
+): (
+	req: http.IncomingMessage,
+	res: http.ServerResponse,
+	next: () => void,
+) => void {
+	_LDObserve._debug('setting up middleware')
+	return async (
+		req: http.IncomingMessage,
+		res: http.ServerResponse,
+		next: () => void,
+	) => {
+		_LDObserve._debug('middleware handling request')
+		return _LDObserve.runWithHeaders(
+			`${req.method?.toUpperCase()} - ${req.url}`,
+			req.headers,
+			async () => {
+				next()
+				await new Promise((resolve) => {
+					res.once('finish', resolve)
+				})
+			},
+			{
+				attributes: {
+					[ATTR_HTTP_REQUEST_METHOD]: req.method,
+					[ATTR_HTTP_ROUTE]: req.url,
+					[ATTR_HTTP_RESPONSE_STATUS_CODE]: res.statusCode,
+					...(metadata ?? {}),
+				},
+			},
+		)
+	}
+}
+
+/**
+ * Express compatible error handler.
+ * Exposed as `Handlers.errorHandler`.
+ * `metadata` accepts structured tags that should be attached to every error.
+ */
+export function errorHandler(
+	options: NodeOptions,
```

Comment:
I think we can change these to probably not take options, or optionally take them? Like have an `_LDObserve._getOpts` or something.